### PR TITLE
Feature/local server component

### LINKED ISSUE #157: Install Server Component from local without fetching from GitHub

### PR DESCRIPTION
**Pull request by worksofliam**
Packages a specific version of the Server Component into the extension vsix, which will then be automatically installed when connected to a server.

Closes #157.

### CODE REVIEW

File: src/connection/serverComponent.ts [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
 
-import { writeFile, unlink } from "fs/promises";
-import os from "os";
+import { stat } from "fs/promises";
+import { SERVER_VERSION_FILE } from "./SCVersion";
 
 const octokit = new Octokit();
 
 // During development, you can set the SERVER_VERSION in .vscode/launch.json
 // Otherwise, fall back to the working version
```
Comment (william-xiang) [inline src/connection/serverComponent.ts:15]: I guess these comments can be removed now?

File: src/extension.ts [hunk 2]
```diff
@@ -28,6 +28,8 @@ export interface Db2i {
 
 export function activate(context: vscode.ExtensionContext): Db2i {
 
+  console.log(context.extensionPath);
```
Comment (william-xiang) [inline src/extension.ts:31]: Not sure if this is added on purpose, seems it's for debugging only.

File: src/connection/serverComponent.ts [hunk 3]
```diff
@@ -90,79 +88,54 @@ export class ServerComponent {
...
 
-        if (lastInstalledName !== basename || this.installed === false) {
-          const updateQuestion = await window.showInformationMessage(
-            `Database Extension update required`, 
-            {
-              modal: true,
-              detail: `An update to the database server component is required: ${basename}`
-            }, 
-            `Update`
-          );
```
Comment (william-xiang) [inline src/connection/serverComponent.ts:125]: So here we will always update the server component if there is update available without letting user decide?

File: src/connection/serverComponent.ts [hunk 4]
```diff
@@ -90,79 +88,54 @@ export class ServerComponent {
...
+            command: `echo ${ExecutablePathDir}`
+          });
 
-              await connection.uploadFiles([{local: tempFile, remote: remotePath}]);
+          if (commandResult.code === 0 && commandResult.stderr === ``) {
+            const remotePath = path.posix.join(commandResult.stdout, basename);
 
-              // Clean up the temp file
-              unlink(tempFile);
+            ServerComponent.writeOutput(JSON.stringify({remotePath, ExecutablePathDir}));
```
Comment (william-xiang) [inline src/connection/serverComponent.ts:120]: I think it's better to use the actual path value from commandResult.

**Review by william-xiang — CHANGES_REQUESTED** [review]
The other two question/suggestion:
1. should the version be bumped for this change?
2. exclude the dsc.ts from the bundle

**Review by edmundreinhardt — APPROVED** [review]

**Review by william-xiang — APPROVED** [review]
Approved. Thank you.

**Comment by worksofliam** [comment]
@william-xiang 

1. We won't do a version bump here because we do not need to.
2. I removed `dsc.ts` from the vsix build.

**Pull request merged**